### PR TITLE
feat(drawer, tabs): add support for custom target overrides via context FE-2888

### PR DESCRIPTION
**Pull request by edleeks87**
### Proposed behaviour

Adds support for being able to override the targeting of the `TabTitle`, meaning it will not display an associated `Tab`'s content but what ever custom target is desired. Also adds mechanism for overriding the validation statuses of the Tabs via the `validationStatusOverrides` prop, this is required as when using custom targeting there is no way of reporting the statuses through the internal mechanism within `Tabs`

### Current behaviour

Currently `TabTitle`s only target their associated `Tab`

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context

image

### Testing instructions

`http://localhost:9001/?path=/story/design-system-drawer--with-tab-controls`

### CODE REVIEW

File: src/components/tabs/__internal__/tab-title/tab-title.style.js [hunk 1]
```diff
@@ -277,7 +277,7 @@ const StyledTabTitle = styled.li`
     ${error || warning || info ? 'border-bottom: none;' : ''}
 
     &:focus {
-      outline: 2px solid ${theme.colors.focus};
+      outline: ${({ hasCustomTarget }) => (hasCustomTarget ? 'none;' : `2px solid ${theme.colors.focus};`)}
```
Comment (dawidzarzycki) [inline src/components/tabs/__internal__/tab-title/tab-title.style.js:280]: I think that you can destructure this `hasCustomTarget`  in line 270

File: src/components/tabs/__internal__/tabs.style.js [hunk 2]
```diff
@@ -6,8 +6,11 @@ const StyledTabs = styled.div`
...
-  ${({ position }) => position === 'left' && css`
-    display: flex;
+  ${({ position, hasCustomTarget }) => position === 'left' && css`
+    ${!hasCustomTarget && css`
+      display: flex;
+    `}
+    
     width: 100%;
     margin-top: 0;
   `}
```
Comment (dawidzarzycki) [inline src/components/tabs/__internal__/tabs.style.js:16]: I suggest to destructure all props at once
```
...
${({ position, theme, hasCustomTarget ... })} => css `
 ... code
`
```

**Comment by dawidzarzycki** [comment]
I can see errors while 
```
npm test src\components\tabs\__internal__\tab\tab.spec.js
```

Also there are two empty folders `tab-title` and `tabs-header`

**Comment by edleeks87** [comment]
> I can see errors while
> 
> ```
> npm test src\components\tabs\__internal__\tab\tab.spec.js
> ```
> 
> Also there are two empty folders `tab-title` and `tabs-header`

Good spot regarding the spec file I forgot to update the mounted component to have the required props, I will update them. As for the additional directories I don't see them, are you sure that isn't just locally on your machine?

**Review by sjohnson-sage — DISMISSED** [review]

**Review by sjohnson-sage — DISMISSED** [review]

**Review by dawidzarzycki — DISMISSED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Pull request merged**